test/jdk/sun/security/ssl/SignatureScheme/SigSchemePropOrdering.java

```diff
@@ -0,0 +1,370 @@
+ * questions.
+ */
+
+//
+// SunJSSE does not support dynamic system properties, no way to re-use
+// system properties in samevm/agentvm mode.
+//
+
+/*
+ * @test
+ * @bug 8255867
+ * @summary SignatureScheme JSSE property does not preserve ordering in handshake messages
+ * @run main/othervm SigSchemePropOrdering
+ */
+
+import java.nio.ByteBuffer;
+import java.util.AbstractMap.SimpleImmutableEntry;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.SSLEngineResult;
+import javax.net.ssl.SSLException;
+
+public class SigSchemePropOrdering {
+
+    // Helper map to correlate integral SignatureScheme identifiers to
+    // their IANA string name counterparts.
+    static final Map<Integer, String> sigSchemeMap = Map.ofEntries(
+            new SimpleImmutableEntry(0x0401, "rsa_pkcs1_sha256"),
+            new SimpleImmutableEntry(0x0501, "rsa_pkcs1_sha384"),
+            new SimpleImmutableEntry(0x0601, "rsa_pkcs1_sha512"),
+            new SimpleImmutableEntry(0x0403, "ecdsa_secp256r1_sha256"),
+            new SimpleImmutableEntry(0x0503, "ecdsa_secp384r1_sha384"),
+            new SimpleImmutableEntry(0x0603, "ecdsa_secp521r1_sha512"),
+            new SimpleImmutableEntry(0x0804, "rsa_pss_rsae_sha256"),
+            new SimpleImmutableEntry(0x0805, "rsa_pss_rsae_sha384"),
+            new SimpleImmutableEntry(0x0806, "rsa_pss_rsae_sha512"),
+            new SimpleImmutableEntry(0x0807, "ed25519"),
+            new SimpleImmutableEntry(0x0808, "ed448"),
+            new SimpleImmutableEntry(0x0809, "rsa_pss_pss_sha256"),
+            new SimpleImmutableEntry(0x080a, "rsa_pss_pss_sha384"),
+            new SimpleImmutableEntry(0x080b, "rsa_pss_pss_sha512"),
+            new SimpleImmutableEntry(0x0101, "rsa_md5"),
+            new SimpleImmutableEntry(0x0201, "rsa_pkcs1_sha1"),
+            new SimpleImmutableEntry(0x0202, "dsa_sha1"),
+            new SimpleImmutableEntry(0x0203, "ecdsa_sha1"),
+            new SimpleImmutableEntry(0x0301, "rsa_sha224"),
+            new SimpleImmutableEntry(0x0302, "dsa_sha224"),
+            new SimpleImmutableEntry(0x0303, "ecdsa_sha224"),
+            new SimpleImmutableEntry(0x0402, "rsa_pkcs1_sha256"));
+
+    // Other useful TLS definitions for these tests
+    private static final int TLS_HS_CLI_HELLO = 1;
+    private static final int TLS_HS_CERT_REQ = 13;
+    private static final int HELLO_EXT_SIG_ALGS = 13;
+
+    // Other definitions we'll need for the test
+    static String STOREPATH = "../../../../javax/net/ssl/etc/";
+    static String KSNAME = "keystore";
+    static String TSNAME = "truststore";
+    static String PASS = "passphrase";
+    private static final String SIG_SCHEME_STR =
+            "rsa_pkcs1_sha256,rsa_pss_rsae_sha256,rsa_pss_pss_sha256," +
+            "ed448,ed25519,ecdsa_secp256r1_sha256";
```

Comment:
It have been a while that we are trying to avoid the use the binary keystore files in test.  It would be nice that if new test cases could use the javax/net/ssl/templates/javax/net/ssl/templates instead.

Reply:
I should be able to rework it into an extension of SSLEngineTemplate.